File: benchmark/c/opl/opl_emu_registers_channel_offset/opl_emu_registers_channel_offset.c
#include <stdio.h>
#include <stdint.h>
#include <stdlib.h>
#include <string.h>

uint32_t opl_emu_registers_channel_offset(uint32_t chnum);
uint32_t opl_emu_registers_channel_offset(uint32_t chnum)
{
    return (chnum % 9) + 0x100 * (chnum / 9);
}

File: flourine_reslut/fail/c/opl/opl_emu_registers_channel_offset/mistral/BaseRepair/opl_emu_registers_channel_offset.rs


use libc::uint32_t;

fn opl_emu_registers_channel_offset(chnum: uint32_t) -> uint32_t {
    let chnum = chnum % 256u32;
    (chnum % 9) + (100 * (chnum / 9)).wrapping_add(0x100)
}

